crates/router/src/connector/noon/transformers.rs

```diff
@@ -67,9 +67,46 @@ pub struct NoonOrder {
     reference: String,
     //Short description of the order.
     name: String,
+    nvp: Option<NoonOrderNvp>,
     ip_address: Option<Secret<String, pii::IpAddress>>,
 }
 
+#[derive(Debug, Serialize)]
+pub struct NoonOrderNvp {
+    #[serde(flatten)]
+    inner: std::collections::BTreeMap<String, Secret<String>>,
+}
+
+fn get_value_as_string(value: &serde_json::Value) -> String {
+    match value {
+        serde_json::Value::Null => "null".to_string(),
+        serde_json::Value::Bool(bool) => bool.to_string(),
+        serde_json::Value::Number(num) => num.to_string(),
+        serde_json::Value::String(string) => string.to_owned(),
+        serde_json::Value::Array(_) => "null".to_string(),
```

Comment:
Ideally merchant should not pass any value as object in the metadata, But if he passes then let's not modify the merchant data to null, Instead pass whatever being passed to us to the connector and let the payment fail at the connector